infra/build_specified_commit.py

```diff
@@ -22,88 +22,93 @@
-                             old_repo_manager=None):
+def build_fuzzers_from_commit(project_name,
+                              commit,
+                              build_repo_manager,
+                              engine='libfuzzer',
+                              sanitizer='address',
+                              architecture='x86_64'):
   """Builds a OSS-Fuzz fuzzer at a  specific commit SHA.
 
   Args:
-    project_name: The OSS-Fuzz project name
-    commit: The commit SHA to build the fuzzers at
-    local_store_path: The full file path of a place where a temp git repo is stored
-    engine: The fuzzing engine to be used
-    sanitizer: The fuzzing sanitizer to be used
-    architecture: The system architiecture to be used for fuzzing
+    project_name: The OSS-Fuzz project name.
+    commit: The commit SHA to build the fuzzers at.
+    build_repo_manager: The OSS-Fuzz project's repo manager to be built at.
+    engine: The fuzzing engine to be used.
+    sanitizer: The fuzzing sanitizer to be used.
+    architecture: The system architiecture to be used for fuzzing.
 
   Returns:
     0 on successful build 1 on failure
   """
-  if not old_repo_manager:
-    inferred_url, repo_name = detect_main_repo_from_docker(project_name, commit)
-    old_repo_manager = repo_manager.RepoManager(
-        inferred_url, local_store_path, repo_name=repo_name)
-  old_repo_manager.checkout_commit(commit)
+  build_repo_manager.checkout_commit(commit)
   return helper.build_fuzzers_impl(
       project_name=project_name,
       clean=True,
       engine=engine,
       sanitizer=sanitizer,
       architecture=architecture,
       env_to_add=None,
-      source_path=old_repo_manager.repo_dir,
-      mount_location=os.path.join('/src', old_repo_manager.repo_name))
+      source_path=build_repo_manager.repo_dir,
+      mount_location=os.path.join('/src', build_repo_manager.repo_name))
 
 
-def detect_main_repo_from_docker(project_name, example_commit, src_dir='/src'):
+def detect_main_repo(project_name, repo_name=None, commit=None, src_dir='/src'):
   """Checks a docker image for the main repo of an OSS-Fuzz project.
 
+  Note: The default is to use the repo name to detect the main repo.
+
   Args:
-    project_name: The name of the OSS-Fuzz project
-    example_commit: An associated commit SHA
-    src_dir: The location of the projects source on the docker image
+    project_name: The name of the oss-fuzz project.
+    repo_name: The name of the main repo in an OSS-Fuzz project.
+    commit: A commit SHA that is associated with the main repo.
+    src_dir: The location of the projects source on the docker image.
 
   Returns:
-    The repo's origin, the repo's name
+    The repo's origin, the repo's name.
   """
+  # TODO: Add infra for non hardcoded '/src'
+  if not repo_name and not commit:
+    print('Error: can not detect main repo without a repo_name or a commit.')
+    return None, None
+
+  if repo_name:
+    helper.build_image_impl('base-builder')
```

Comment:
Add a comment to say why is this needed here only when `repo_name` is specified?

Reply:
Done.